src/Tribe/Tickets_View.php

```diff
@@ -0,0 +1,516 @@
+<?php
+// don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+class Tribe__Tickets__Tickets_View {
+
+	/**
+	 * Get (and instantiate, if necessary) the instance of the class
+	 *
+	 * @static
+	 * @return self
+	 *
+	 */
+	public static function instance() {
+		static $instance;
+
+		if ( ! $instance instanceof self ) {
+			$instance = new self;
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Hook the necessary filters and Actions!
+	 *
+	 * @static
+	 * @return self
+	 */
+	public static function hook() {
+		$myself = self::instance();
+		$main = Tribe__Tickets__Main::instance();
+
+		add_action( 'template_redirect', array( $myself, 'authorization_redirect' ) );
+		add_action( 'template_redirect', array( $myself, 'update_tickets' ) );
+
+		// Generate Non TEC Permalink
+		add_action( 'generate_rewrite_rules', array( $myself, 'add_non_event_permalinks' ) );
+		add_filter( 'query_vars', array( $myself, 'add_query_vars' ) );
+		add_filter( 'the_content', array( $myself, 'intercept_content' ) );
+
+		// Only Applies this to TEC users
+		if ( class_exists( 'Tribe__Events__Rewrite' ) ) {
+			add_action( 'tribe_events_pre_rewrite', array( $myself, 'add_permalink' ) );
+			add_filter( 'tribe_events_rewrite_base_slugs', array( $myself, 'add_rewrite_base_slug' ) );
+		}
+
+		// Intercept Template file for Tickets
+		add_action( 'tribe_events_template', array( $myself, 'intercept_template' ), 20, 2 );
+
+		// We will inject on the Priority 4, to be happen before RSVP
+		add_action( 'tribe_events_single_event_after_the_meta', array( $myself, 'inject_link_template' ), 4 );
+		add_filter( 'the_content', array( $myself, 'inject_link_template_the_content' ) );
+
+		return $myself;
+	}
+
+
+	/**
+	 * For non events the links will be a little bit weird, but it's the safest way
+	 *
+	 * @param WP_Rewrite $wp_rewrite
+	 */
+	public function add_non_event_permalinks( WP_Rewrite $wp_rewrite  ) {
+		$rules = array(
+			'tickets/([0-9]{1,})/?' => 'index.php?p=$matches[1]&tribe-edit-orders=1',
+		);
+
+		$wp_rewrite->rules = $rules + $wp_rewrite->rules;
+	}
+
+	/**
+	 * Add a new Query Var to allow tickets editing
+	 * @param array $vars
+	 * @return array
+	 */
+	public function add_query_vars( $vars ) {
+		$vars[] = 'tribe-edit-orders';
+		return $vars;
+	}
+
+	/**
+	 * Update the RSVP and Tickets values for each Attendee
+	 */
+	public function update_tickets() {
+		// Now fetch the display and check it
+		$display = get_query_var( 'eventDisplay', false );
+		if ( 'tickets' !== $display ) {
+			return;
+		}
+
+		if ( empty( $_POST['process-tickets'] ) || empty( $_POST['attendee'] ) ) {
+			return;
+		}
+
+		$event_id = get_the_ID();
+		$attendees = $_POST['attendee'];
+
+		foreach ( $attendees as $order_id => $data ) {
+			/**
+			 * An Action fired after each Ticket/RSVP is Updated
```

Comment:
So if I'm understanding things correctly this could be slightly misleading - it doesn't seem to fire _after_ the ticket is updated, rather the ticket is updated _when_ this fires. Depending on the order things are registered, even if a dev uses the default priority their own callback could run _before_ the ticket/rsvp is updated.